src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/data/CompareUtil.java

```diff
@@ -122,4 +125,41 @@ public char getOperator() {
             return '~';
         }
     }
+
+    /**
+     * Compares just like plain comparison, however, patterns are expanded using the
+     * given system environment settings.
+     */
+    class PlainVarCompareUtil implements CompareUtil {
```

Comment:
I didn't want to mess too much with the existing code, but this would definitely be a good improvement. I can do that.